app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperManager.kt

```diff
@@ -92,9 +103,41 @@ class WallpaperManager(private val settings: Settings) {
     private fun getCurrentWallpaperFromSettings(): Wallpaper {
         val currentWallpaper = settings.currentWallpaper
         return if (currentWallpaper.isEmpty()) {
-            Wallpaper.NONE
+            defaultWallpaper
+        } else {
+            availableWallpapers.find { it.name == currentWallpaper } ?: defaultWallpaper
+        }
+    }
+
+    fun loadWallpaperFromAssets(wallpaper: Wallpaper, context: Context): Bitmap {
+        val path = if (isLandscape(context)) {
+            wallpaper.landscapePath
         } else {
-            Wallpaper.valueOf(currentWallpaper)
+            wallpaper.portraitPath
+        }
+        return context.assets.open(path).use {
+            BitmapFactory.decodeStream(it)
         }
     }
+
+    private fun isLandscape(context: Context): Boolean {
+        return context.resources.configuration.orientation == Configuration.ORIENTATION_LANDSCAPE
+    }
+
+    fun loadWallpapers() {
+        availableWallpapers.clear()
+        availableWallpapers.add(defaultWallpaper)
+        availableWallpapers.addAll(wallpaperStorage.all())
+        currentWallpaper = getCurrentWallpaperFromSettings()
+    }
```

Comment:
Alternatively to the above comment, you could change the property `availableWallpapers` property to be mutable, since we overwrite the list every time it changes anyway:

```
var availableWallpapers = listOf(defaultWallpaper)
    private set

fun loadWallpapers() {
    availableWallpapers = listOf(defaultWallpaper, *wallpaperStorage.all())
}
```